projects/golang/go/Makefile

```diff
@@ -107,7 +143,7 @@ clean:
 
 .PHONY: install-deps
 install-deps:
-	yum install -y which git rpmdevtools go-srpm-macros awscli rpm-build pcre-devel glibc-static hostname procps-ng tar
```

Comment:
now that we're using the builder-base as the base image for the EKS Go prow jobs we no longer need to install so many dependencies during the build. The `yum-plugin-ovl` is required for installing what dependencies we do have on the overlay fs.